flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/FrontMetricGroup.java

```diff
@@ -30,28 +34,54 @@
  */
 public class FrontMetricGroup<P extends AbstractMetricGroup<?>> extends ProxyMetricGroup<P> {
 
-	protected int reporterIndex;
+	private final ReporterIndexInjectingMetricScope scope;
 
 	public FrontMetricGroup(int reporterIndex, P reference) {
 		super(reference);
-		this.reporterIndex = reporterIndex;
+		this.scope = new ReporterIndexInjectingMetricScope(reporterIndex, this.parentMetricGroup.getScope());
 	}
 
 	@Override
 	public String getMetricIdentifier(String metricName) {
-		return parentMetricGroup.getMetricIdentifier(metricName, null, this.reporterIndex);
+		return scope.getMetricIdentifier(metricName);
 	}
 
 	@Override
 	public String getMetricIdentifier(String metricName, CharacterFilter filter) {
-		return parentMetricGroup.getMetricIdentifier(metricName, filter, this.reporterIndex);
+		return scope.getMetricIdentifier(metricName, filter);
 	}
 
 	public String getLogicalScope(CharacterFilter filter) {
 		return parentMetricGroup.getLogicalScope(filter);
 	}
 
 	public String getLogicalScope(CharacterFilter filter, char delimiter) {
-		return parentMetricGroup.getLogicalScope(filter, delimiter, this.reporterIndex);
+		return parentMetricGroup.getLogicalScope(filter, delimiter);
```

Comment:
**NB:** If I understand correctly, without `reporterIndex`, you are making this call non cached.
This would impact InfluxDB, Jmx and Prometheus reporters.

Reply:
This was unintentional, reverting.